backends/platform/ios7/ios7_video.mm

```diff
@@ -535,9 +535,19 @@ -(void)adjustViewFrameForSafeArea {
 			newFrame = CGRectMake(screenSize.origin.x + inset.left, screenSize.origin.y, screenSize.size.width - inset.left, height);
 		}
 
-		// The onscreen control buttons have to be moved accordingly
-		[_menuButton setFrame:CGRectMake(newFrame.size.width - _menuButton.imageView.image.size.width, 0, _menuButton.imageView.image.size.width, _menuButton.imageView.image.size.height)];
-		[_toggleTouchModeButton setFrame:CGRectMake(newFrame.size.width - _toggleTouchModeButton.imageView.image.size.width - _toggleTouchModeButton.imageView.image.size.width, 0, _toggleTouchModeButton.imageView.image.size.width, _toggleTouchModeButton.imageView.image.size.height)];
+		// Add margins to respect the iPhone and iPad safe areas. Use the right
+		// safe area inset if available since the position of the buttons are
+		// on the right hand side. The iPhone corners on the later models have
+		// the form of squircles rather than circles which cause button images
+		// to become cropped if placed too close to the corner.
+		// iPads has a more rectangular screen form. The inset margin on iPads
+		// is 0, however due to the rounding of the corners images gets cropped
+		// if placed too close to the corners. Use a constant margin for those.
+		const CGFloat margin = inset.right > 0 ? inset.right/4 : 10;
+		// Touch mode button on top
+		[_toggleTouchModeButton setFrame:CGRectMake(self.frame.size.width - _toggleTouchModeButton.imageView.image.size.width - margin, margin, _toggleTouchModeButton.imageView.image.size.width, _toggleTouchModeButton.imageView.image.size.height)];
+		// Burger menu button below
+		[_menuButton setFrame:CGRectMake(self.frame.size.width - _menuButton.imageView.image.size.width - margin, _toggleTouchModeButton.imageView.image.size.height + margin, _menuButton.imageView.image.size.width, _menuButton.imageView.image.size.height)];
```

Comment:
Very minor comment: this code does a right-align for the two images (their right border is aligned). I am wondering if a horizontal center alignment might in theory be better. However I am assuming the two images have the same size (or at least the same width), in which case it does not matter in practice.

Reply:
Yes, good point. The images do have the same sizes. Could still be fixed in case the images are changed in the future. 

I also noticed a bug with hiding the burger menu button when in a game and from settings changing to not show on screen buttons. 

Will fix this.

Reply:
I must have been using an old build when I noticed the bug. Everything works as expected.